tagcompare/main.py

```diff
@@ -1,10 +1,108 @@
+
+    The "answer" return value is True for "yes" or False for "no".
+
+    Source: http://code.activestate.com/recipes/577058/
+    """
+    valid = {"yes": True, "y": True, "ye": True,
+             "no": False, "n": False}
+    if default is None:
+        prompt = " [y/n] "
+    elif default == "yes":
+        prompt = " [Y/n] "
+    elif default == "no":
+        prompt = " [y/N] "
+    else:
+        raise ValueError("invalid default answer: '%s'" % default)
+
+    while True:
+        sys.stdout.write(question + prompt)
+        choice = raw_input().lower()
+        if default is not None and choice == '':
+            return valid[default]
+        elif choice in valid:
+            return valid[choice]
+        else:
+            sys.stdout.write("Please respond with 'yes' or 'no' "
+                             "(or 'y' or 'n').\n")
+
+
+def __parse_params_to_settings():
+    """
+    Handles parsing commandline args and set appropriate settings from them
+    :return: the key/value pair for the commandline args parsed
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--capture-only', action='store_true', default=False,
+                        help='Run capture only without compare')
+
+    parser.add_argument('-d', '--domain',
+                        default=None,
+                        help='Domain, i.e. www.placelocal.com')
+
+    parser.add_argument('-v', '--verbose', action='store_true', default=False,
+                        help='Enable verbose logging for debugging')
+
+    # Either campaigns or publishers need to be specified, but not both
+    group = parser.add_mutually_exclusive_group(required=False)
+    group.add_argument('-c', '--campaigns', nargs='+',
+                       help='Space separated list of campaign ids')
+    group.add_argument('-p', '--publishers', nargs='+',
+                       help='Space separated list of publisher ids')
+    args = parser.parse_args()
+    LOGGER.debug("tagcompare params: %s", args)
+    return args
+
+
+def __update_settings_from_args(args):
+    # Update settings based on params
+    if args.campaigns or args.publishers:
+        print args
+        settings.DEFAULT.campaigns = args.campaigns
+        settings.DEFAULT.publishers = args.publishers
+
+    if args.domain:
+        settings.DEFAULT.domain = args.domain
+
+    if args.verbose:
+        settings.DEFAULT.loglevel = logging.DEBUG
 
 
 def main():
+    args = __parse_params_to_settings()
+    __update_settings_from_args(args)
+
+    proceed = __query_yes_no("Start tagcompare against {} for cid={}, pids={}?"
+                             .format(settings.DEFAULT.domain,
+                                     settings.DEFAULT.campaigns,
+                                     settings.DEFAULT.publishers))
+
+    if not proceed:
+        print("Stopping tagcompare on user input")
+        exit(0)
+
     jobname = capture.main()
-    compare.main(build=jobname)
+    if not args.capture_only:
```

Comment:
Preferential but I'd simplify this to make the argument --run-compare that way it's easier to read